warehouse/query-core/src/test/java/datawave/query/jexl/visitors/IsNotNullPruningVisitorTest.java

```diff
@@ -0,0 +1,112 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.exceptions.InvalidQueryTreeException;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.visitors.validate.ASTValidator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class IsNotNullPruningVisitorTest {
+    
```

Comment:
More of a question for my curiosity, but does it make sense to have a test case to see what occurs if the IsNotNull queries are the only ones in the string, to see if the scenario is handled as would be expected?

Reply:
Sounds worthwhile to me, I'll add a few more test cases to handle what you describe